compiler/crates/relay-codegen/src/build_ast.rs

```diff
@@ -744,15 +849,41 @@ impl<'schema, 'builder> CodegenBuilder<'schema, 'builder> {
             CODEGEN_CONSTANTS.relay_resolver
         };
 
-        // TODO(T86853359): Support non-haste environments when generating Relay Resolver Reader AST
-        let haste_import_name = Path::new(&module.to_string())
-            .file_stem()
-            .unwrap()
-            .to_string_lossy()
-            .intern();
-
         let args = self.build_arguments(field_arguments);
 
+        let definition_artifact_location = path_for_artifact(
+            self.project_config,
+            match definition {
+                ExecutableDefinition::Operation(operation) => {
+                    operation.name.location.source_location()
+                }
+                ExecutableDefinition::Fragment(fragment) => {
+                    fragment.name.location.source_location()
+                }
+            },
+            match definition {
+                ExecutableDefinition::Operation(operation) => operation.name.item,
+                ExecutableDefinition::Fragment(fragment) => fragment.name.item,
+            },
+        );
+
+        let module_location = PathBuf::from_str(module.lookup()).unwrap();
+
+        let module_path = module_location.parent().unwrap();
```

Comment:
Why do we need `.parent()` here?

Reply:
this `pathdiff` library strictly works for directories. If you don't strip the file off the end, we'll end up with some funky (incorrect) paths

Reply:
I'll leave a comment inline about this